src/selectors/columnWidths.js

```diff
@@ -50,31 +54,37 @@ function columnWidths(
   const scrollbarSpace = scrollEnabledY ? scrollbarYWidth : 0;
   const viewportWidth = width - scrollbarSpace;
 
-  const { newColumnGroupProps, newColumnProps } = flexWidths(
-    columnGroupProps,
-    columnProps,
-    viewportWidth
-  );
+  const {
+    columnGroupProps: columnGroupPropsWithFlex,
+    columnProps: columnPropsWithFlex,
+  } = flexWidths(columnGroupProps, columnProps, viewportWidth);
   const {
     fixed: fixedColumns,
     fixedRight: fixedRightColumns,
     scrollable: scrollableColumns,
-  } = groupElements(newColumnProps);
+  } = groupElements(columnPropsWithFlex);
   const {
     fixed: fixedColumnGroups,
     fixedRight: fixedRightColumnGroups,
     scrollable: scrollableColumnGroups,
-  } = groupElements(newColumnGroupProps);
+  } = groupElements(columnGroupPropsWithFlex);
 
   const availableScrollWidth =
     viewportWidth -
     getTotalWidth(fixedColumns) -
     getTotalWidth(fixedRightColumns);
-  const maxScrollX = Math.max(0, getTotalWidth(newColumnProps) - viewportWidth);
+  const maxScrollX = Math.max(
+    0,
+    getTotalWidth(columnPropsWithFlex) - viewportWidth
+  );
 
   return {
-    columnGroupProps: newColumnGroupProps,
-    columnProps: newColumnProps,
+    columnGroupProps: concat(
+      fixedColumnGroups,
+      scrollableColumnGroups,
+      fixedRightColumnGroups
+    ),
+    columnProps: concat(fixedColumns, scrollableColumns, fixedRightColumns),
```

Comment:
`columnGroupProps` and `columnProps` are now correctly sorted/partitioned based on cell group type.